### ISSUE TITLE
Unable to connect to device.

### ISSUE THREAD

**Issue by jpuerto** [pauldemarco/flutter_blue#440]
We are experiencing an issue connecting to the device after many disconnect/connect actions. We have a screen in the application that needs a connection with a BT device, a call to disconnect is made when we leave this screen. Switching fast the screens to stress the application leads into the following errors:

```
2019-11-13 13:25:07.426 1976-2039/? W/bt_stack: [WARNING:bta_gattc_act.cc(1039)] bta_gattc_conn_cback: cif=3 connected=0 conn_id=0x0003 reason=0x0016
2019-11-13 13:25:07.426 1976-2039/? W/bt_stack: [WARNING:bta_gattc_act.cc(1039)] bta_gattc_conn_cback: cif=4 connected=0 conn_id=0x0004 reason=0x0016
2019-11-13 13:25:07.426 1976-2039/? W/bt_stack: [WARNING:bta_gattc_act.cc(1039)] bta_gattc_conn_cback: cif=5 connected=0 conn_id=0x0005 reason=0x0016
```
We have found the following issue describing the same problem that we have.

* https://lists.w3.org/Archives/Public/public-web-bluetooth-log/2016Jan/0029.html
* https://lists.w3.org/Archives/Public/public-web-bluetooth-log/2016Jan/0033.html
* https://bugs.chromium.org/p/chromium/issues/detail?id=576819

Basically, the bluetooth has not enough free resources to perform more connections, reviewing the code of FlutterBlue, it creates a new BluetoothGatt object with every connection and calls to close it on disconnect only if the device status is disconnected. Somehow, this is causing the problem described in the first link.
